apps/ui/src/composables/useSharing.ts

```diff
@@ -0,0 +1,82 @@
+import { FunctionalComponent } from 'vue';
+import { Proposal } from '@/types';
+import ICFarcaster from '~icons/c/farcaster';
+import ICLens from '~icons/c/lens';
+import ICX from '~icons/c/x';
+
+export type ShareableType = 'proposal' | 'user';
+type SocialNetwork = 'x' | 'lens' | 'farcaster';
+
+const HASH_TAG = 'Snapshot';
+
+const SOCIAL_NETWORKS: {
+  id: SocialNetwork;
+  name: string;
+  icon: FunctionalComponent;
+}[] = [
+  { id: 'x', name: 'X', icon: ICX },
+  { id: 'lens', name: 'Hey', icon: ICLens },
+  { id: 'farcaster', name: 'Warpcast', icon: ICFarcaster }
+];
+
+export function useSharing() {
+  function getProposalUrl(proposal: Proposal): string {
+    return `https://${window.location.hostname}/#/${proposal.network}:${proposal.space.id}/proposal/${proposal.proposal_id}`;
```

Comment:
I don't think it is required but it is missing port here
<img width="631" alt="Untitled 4" src="https://github.com/user-attachments/assets/d99cfa75-7975-491d-96c9-b2075ac998b6">

Reply:
This should not be an issue for anything else than localhost

Reply:
I don't gett why we do this separately? we can do same as User right? I mean something like `encodeURIComponent(window.location.href)` what's different with proposal URL?

Reply:
Don't get either, this refactor is just moving the existing logic to a composable

Reply:
Maybe also port that to the other to `User` and `SpaceUser`, for consistency

Reply:
And maybe we could build that with vuejs router, instead of manually

Reply:
I feel like it is better to use vuejs router and keep it same everywhere for consistency, also I feel like we can use page title instead of passing data from all components 

<img width="354" alt="image" src="https://github.com/user-attachments/assets/ea355f71-c437-474f-8f65-de7293b21a4e">

Reply:
What do you mean by using page title?

Reply:
All url are now computed with vuejs router

Reply:
> I feel like it is better to use vuejs router and keep it same everywhere for consistency, also I feel like we can use page title instead of passing data from all components
> 
> <img alt="image" width="354" src="https://private-user-images.githubusercontent.com/15967809/358314837-ea355f71-c437-474f-8f65-de7293b21a4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3Nzk1NzQsIm5iZiI6MTcyMzc3OTI3NCwicGF0aCI6Ii8xNTk2NzgwOS8zNTgzMTQ4MzctZWEzNTVmNzEtYzQzNy00NzRmLThmNjUtZGU3MjkzYjIxYTRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE2VDAzMzQzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwMjRlN2FkMzZiMWJiMmE1N2M2MjNkYjg2NmQ0MWVkMDY3Y2UzMjZiNjVmOGE4M2I0ZDkxOTgwZjQyMmY2NDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rdQ9GGIlZ2q96zp7bXA4uPHySKYN8k_NPZFF9rstHHI">

For the upcoming share vote, the message will be more customized, like: "I just voted XXX on PROPOSAL-NAME"

Reply:
> For the upcoming share vote, the message will be more customized, like: "I just voted XXX on PROPOSAL-NAME"

Ahh cool then 😅